src/openms/include/OpenMS/METADATA/ID/ScoredProcessingResult.h

```diff
@@ -0,0 +1,117 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2018.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Hendrik Weisser $
+// $Authors: Hendrik Weisser $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_METADATA_ID_SCOREDPROCESSINGRESULT_H
+#define OPENMS_METADATA_ID_SCOREDPROCESSINGRESULT_H
+
+#include <OpenMS/METADATA/ID/DataProcessingStep.h>
+#include <OpenMS/METADATA/ID/ScoreType.h>
+
+namespace OpenMS
+{
+  namespace IdentificationDataInternal
+  {
+    /// Base class for ID data with scores and processing steps (and meta info)
+    struct ScoredProcessingResult: public MetaInfoInterface
+    {
+      ScoreList scores;
```

Comment:
I dont think that this should be a public member, please make something like this private and give access through a clean interface (e.g. getScore / setScore)

that basically translates to `std::vector<std::pair< IteratorWrapper<std::set<ScoreType> ::iterator> , double>> ScoreList;`

Reply:
This now supports multiple scores, so I would argue that the list of scores *is* a clean interface. I don't agree with writing unnecessary getters/setters if there are no invariants to check.

I am however still thinking about changing the implementation behind `ScoreList` (e.g. to directly include the processing step that assigned a score, or to allow efficient retrieval by `ScoreType` and in order via `boost::multi_index_container`).

Reply:
> I am however still thinking about changing the implementation behind `ScoreList` (e.g. to directly include the processing step that assigned a score, or to allow efficient retrieval by `ScoreType` and in order via `boost::multi_index_container`).

having a simple, clean interface would make it easier to change the underlying implementation. Also, it is not possible to wrap something like `std::vector<std::pair< IteratorWrapper<std::set<ScoreType> ::iterator> , double>> ` in Python. 

> I don't agree with writing unnecessary getters/setters if there are no invariants to check.

I agree for simple POD. However, if the implementation is complex and subject to change, it can make sense. Also, unless the implementation is POD / a trivial member variable, we will need that for Python

Reply:
Let's discuss these interface questions in our upcoming call.